tools/redis-tools.sh

```diff
@@ -0,0 +1,125 @@
+# Redis. There are 5 main functions: keys, values, delete, dump and restore.
+# The first five will query over Redis to find keys that have the same value
+# in a specific field, as determined by the passed arguments. Restore will
+# restore all keys dumped in a directory. If "all" is used in [Field Name]
+# and [Field Value] is left empty, all keys will be selected. 
+
+usage() {
+  cat << EOF
+  Usage:
+    ./redis-tools.sh keys [Field Name] [Field Value]
+        The matching keys are printed out.
+    ./redis-tools.sh values [Field Name] [Field Value]
+        The values of the matching keys are printed out.
+    ./redis-tools.sh delete [Field Name] [Field Value]
+        The matching keys are deleted from redis.
+    ./redis-tools.sh dump [Field Name] [Field Value] [Directory to Dump]
+        The matching keys are dumped to the indicated directory.
+    ./redis-tools.sh restore [Directory to Restore] 
+        The keys dumped in the directory are restored.
+
+  Examples: 
+  ./redis-tools.sh keys all
+  ./redis-tools.sh values id f5bd194d9424427c9d82a57eedc531d0
+  ./redis-tools.sh dump request_id 09879dbc90a44c6db80dc9f68113d8 storagefolder
+  ./redis-tools.sh delete all
+  ./redis-tools.sh restore storagefolder
+  ./redis-tools.sh dump all storagefolder
+EOF
+}
+
+action=$1 # "keys" / "values" / "delete" / "dump" / "restore"
+
+case "$action" in
+  "restore")
+    directory=$2
+    if [[ -z "$directory" ]]; then
+      >&2 echo Error: Directory not provided
+      exit
+    fi
+    for filepath in "$directory"/*; do
+      key=$(basename "$filepath")
+      echo Restoring "$key"
+      redis-cli -x restore "$key" 0 < "$filepath"
+    done
+  ;;
+
+  "keys"|"values"|"dump"|"delete")
+    field_name="$2"
+    # Ignores [Field Value] if "all" is passed
+    if [[ "$field_name" == "all" ]]; then
+      directory="$3";
+    else
+      field_value="$3"
+      directory="$4"
+    fi
+    if [[ "$action" == "dump" ]] && [[ -z "$directory" ]]; then
+      >&2 echo Error: Directory not provided
+      exit
+    fi
+    if [[ "$action" != "values" ]]; then
+      declare -a key_array
+      if [[ "$action" == keys ]] || [[ "$field_name" != "all" ]]; then
+        echo Keys found:
+      fi
+    fi
+    for key in $(redis-cli --scan); do 
+      # Gets the key_type by removing the characters after the colon 
+      key_type=$(echo "$key" | cut -f 1 -d :)
+      if [[ "$key_type" != "TurbiniaTask" ]]; then
+        continue
+      fi
+      # Gets the Task value and split its key:value pairs into an array
+      value=$(redis-cli get "$key")
+      modified_value="${value//\"/}"
+      modified_value="${modified_value//\{/}"
+      modified_value="${modified_value//\}/}"
+      IFS=',' read -r -a array <<< "$modified_value"
+      if [[ "$field_name" == "all" ]]; then
+        if [[ "$action" == "values" ]]; then
+          echo -e "$value\n"
+        else
+          if [[ "$action" == keys ]]; then echo "$key"; fi
```

Comment:
If I understand this correctly, we are printing every key here. I can see that being a problem if there are thousands of keys, which is quite possible. Do we need to print every key? Maybe instead we can print a count of keys that were found?

Reply:
Another variant could be to show the count only if there are more than N keys (50?).

Reply:
(Will wait for the outcome of this comment before merging)

Reply:
I had replaced they keys with count for delete and dump, but since keys is meant to print the keys found, and value is meant to print the values, I had decided to leave them that way with all, so that it is possible to print all keys/values if necessary, even if it takes long. But I just made a new function called count, to simply print how many keys are stored or all keys that match the arguments. Let me know if that works.